plugins/services/src/js/utils/__tests__/DeclinedOffersUtil-test.js

```diff
@@ -0,0 +1,132 @@
+jest.dontMock('../DeclinedOffersUtil');
+
+const DeclinedOffersUtil = require('../DeclinedOffersUtil');
+const QueueFixture = require('./fixtures/QueueFixture.json');
+
+describe('DeclinedOffersUtil', function () {
+
+  describe('#getSummaryFromQueue', function () {
+
+    it('returns null when processedOffersSummary undefined', function () {
+      expect(DeclinedOffersUtil.getSummaryFromQueue({})).toEqual(null);
+    });
+
+    it('returns null when unusedOffersCount is 0', function () {
+      expect(DeclinedOffersUtil.getSummaryFromQueue({
+        processedOffersSummary: 0
+      })).toEqual(null);
+    });
+
+    it('returns null when lastUnusedOffers is undefined, null, or an empty array', function () {
+      expect(DeclinedOffersUtil.getSummaryFromQueue({})).toEqual(null);
+      expect(DeclinedOffersUtil.getSummaryFromQueue({
+        lastUnusedOffers: null
+      })).toEqual(null);
+      expect(DeclinedOffersUtil.getSummaryFromQueue({
+        lastUnusedOffers: []
+      })).toEqual(null);
+    });
+
+    it('cascades matched offers from roles to subsequent offers', function () {
+      const summary = DeclinedOffersUtil.getSummaryFromQueue({
+        app: {},
+        processedOffersSummary: {
+          processedOffersCount: 10,
+          unusedOffersCount: 1,
+          rejectReason: {
+            UnfulfilledRole: 1,
+            UnfulfilledConstraint: 0,
+            InsufficientCpus: 0,
+            InsufficientMemory: 0,
+            InsufficientDisk: 0,
+            InsufficientPorts: 0
+          }
+        },
+        lastUnusedOffers: [
+          {
+            foo: 'bar'
+          }
+        ]
+      });
+
+      expect(summary.role.matched).toEqual(9);
+      expect(summary.constraint.offers).toEqual(9);
+      expect(summary.constraint.matched).toEqual(9);
+      expect(summary.cpu.offers).toEqual(9);
+      expect(summary.cpu.matched).toEqual(9);
+      expect(summary.mem.offers).toEqual(9);
+      expect(summary.mem.matched).toEqual(9);
+      expect(summary.disk.offers).toEqual(9);
+      expect(summary.disk.matched).toEqual(9);
+      expect(summary.ports.offers).toEqual(9);
+      expect(summary.ports.matched).toEqual(9);
+    });
+
+    it('transforms the API response into the expected format', function () {
+      const summary = DeclinedOffersUtil.getSummaryFromQueue(QueueFixture.queue[1]);
```

Comment:
It would be really nice if we didn't use the fixtures for testing and specified a slice of the fixture here in the test. That way test code is tightly coupled to the fields it expects as inputs and the fields it returns as an output. It's also a little more future proof because these fixtures might disappear with GraphQL. I also think it's more readable to look at the tests and see what the example input looks like. What do you think?